To Do list: add & remove

jlberbesi opened the pull request:
For this milestone I have:

- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.

jlberbesi committed:
Add externalFuntions.js and updates in index.js

jlberbesi committed:
Add code reviewer corrections2

jlberbesi committed:
Add final actions for project

Davidosky007 reviewed (commented):
Hi @jlberbesi 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong PR for this milestone `To Do list: interactive list` kindly submit the correct PR that follows [This requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

jlberbesi commented:
> Hi @jlberbesi 👋,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> You have submitted the wrong PR for this milestone `To Do list: interactive list` kindly submit the correct PR that follows [This requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md)
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

@Davidosky007  Hi thanks a lot for your review. Sorry about the confusion here. However ,I would like to highlight I have implemented a new script here called promoteFunctions.js and also the index.js file had few changes as well to be aligned with the requirements for this project.  I think I pasted the wrong link in the dashboard.

BenMukebo commented:
Hello @jlberbesi !!! I did not catch you well here?
@Davidosky007 Hi thanks a lot for your

>  review. Sorry about the confusion here. However ,I would like to highlight I have implemented a new script here called promoteFunctions.js and also the index.js file had few changes as well to be aligned with the requirements for this project. I think I pasted the wrong link in the dashboard.

jlberbesi commented:
Hello everyone! I wanted to share a brief video explanation of our latest milestone and demonstrate how it's functioning on my end. 

[My loom video](https://www.loom.com/share/c6e17f7a61b944b2afa155de6444f9c5)

BenMukebo commented:
@jlberbesi, If it's a to-do list: add and remove a milestone not an interactive list, and go invalidate that PR so you can resubmit it using the correct submit button. Hope this makes sense

jlberbesi commented:
> Hello @jlberbesi !!! I did not catch you well here? @Davidosky007 Hi thanks a lot for your
> 
> > review. Sorry about the confusion here. However ,I would like to highlight I have implemented a new script here called promoteFunctions.js and also the index.js file had few changes as well to be aligned with the requirements for this project. I think I pasted the wrong link in the dashboard.

@BenMukebo Hello. I apologize if my message was not clear. As you know, English is not my first language. I mentioned that I found it strange that error occurred, as I had created this new branch and added some changes to it. I also suggested that perhaps I had copied an incorrect link in the dashboard

BenMukebo reviewed (commented):
...

jlberbesi commented:
@BenMukebo  Here is the picture with details you asked me for.
<img width="1096" alt="Screenshot 2023-04-09 at 12 15 32 AM" src="https://user-images.githubusercontent.com/88105477/230755698-246563eb-ba78-40c7-bde5-65d329044c4f.png">

BenMukebo reviewed (approved):
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @jlberbesi**, 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- It has one new JavaScript files imported as a modul.
- It has event listener on the checkbox. ✔️ 
- The task object's values for `completed` is updated upon user actions. ✔️ 
-  A function for the "Clear all completed" button using the `Array.filter()` method is implemented. ✔️
- The updated array of items is stored in local storage. ✔️  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

emmanuellekamwa reviewed (approved):
**STATUS: APPROVED 🍾✔🎉**
Hello @jlberbesi 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
